### ISSUE TITLE
Error in month count when continuous use nextMonth or prevMonth 

### ISSUE THREAD

**Issue by PlayApple** [malcommac/SwiftDate#449]
When I use prevMonth and prevMonth, the month count calculate mistake.
See the log pic:

![2017-07-13 12 18 21](https://user-images.githubusercontent.com/3360969/28150207-bafec8c6-67c5-11e7-9d10-164f6eeb4fb2.png)

**Comment by PlayApple** [comment]
I want to implement switching time periods. If nextMonth is wrong, I'll have to do it some other way.

![2017-07-13 12 30 15](https://user-images.githubusercontent.com/3360969/28150463-4c96f730-67c7-11e7-808e-bdb46710233c.png)

**Comment by malcommac** [comment]
Since 4.1.7 `nextMonth` and `prevMonth` properties are deprecated and replaced by:

(for `DateInRegion`):
```swift
- public func prevMonth(at time: TimeReference = .auto) -> DateInRegion
- public func nextMonth(at time: TimeReference = .auto) -> DateInRegion
```
and (for plain `Date`):
```swift
- public func prevMonth(at time: TimeReference = .auto) -> Date
- public func nextMonth(at time: TimeReference = .auto) -> Date
```

`TimeReference` allows you to specify the way in which the date is calculated:
* `.auto`: evaluated date is calculated by adding one time component (month/week) to the current date.
* `.start`: result date is the first day of the next time component (month/week) (at 00:00:00).
* `.end`: result date is the last day of the next time component (month/week) (at 23:59:59).

So, coming back to your issue you can get the first day of each month by using:

```swift
let gmt = Region(tz: TimeZoneName.gmt, cal: CalendarName.gregorian, loc: LocaleName.english)
llet startDate = DateInRegion(string: "2017-07-16 16:00:00 +0000", format: .iso8601Auto, fromRegion: gmt)!
var nextDate = startDate.prevMonth(at: .start) // move at the start of previous month
for _ in 0..<10 {
  nextDate = nextDate.nextMonth(at: .start) // get the first day of the next month (use .end to get the last day)
  print(nextDate.string(format: .iso8601Auto))
}
```

I've also added `prevWeek()` and `nextWeek()` both to `Date` and `DateInRegion`.

Thank you for your report.
These fixes will be included in 4.1.7.

**Comment by malcommac** [comment]
Reference commits:
- https://github.com/malcommac/SwiftDate/commit/c500dfa97e2497a0f24cf9ae579deb93dabcfb38 for `nextMonth()` and `prevMonth()`
- https://github.com/malcommac/SwiftDate/commit/4ccc2960caaf86cae96b84c4e9857edd1f017c9c for `nextWeek()` and `prevWeek()`

and finally https://github.com/malcommac/SwiftDate/commit/d6bbfd022c867250176f1b1cf3437741be9bb5cb for unit tests.
